### PR TITLE
Typedocs CICD

### PR DESCRIPTION
**Pull request by jacobcreech**
**Problem**:
Typedocs are not generated on the fly for each js/ts library under spl. They end up manually generated or lagged behind the actual releases

**Solution**:
Created a CICD process using actions to generate typedocs for each of the js/ts bindings we have under spl. Docs will be deployed on release.

You can see an example of CICD here: https://github.com/Jacob-CG/solana-program-library/runs/4051236499?check_suite_focus=true

Currently js docs are using [documentation.js](https://documentation.js.org/)

### CODE REVIEW

File: token-lending/js/package.json [hunk 1]
```diff
@@ -23,8 +23,7 @@
     ],
     "scripts": {
         "build": "rm -rf lib/* && tsc && rollup -c && rm lib/rollup.config.d.ts",
-        "deploy": "yarn docs && gh-pages -d docs -e token-lending",
-        "docs": "rm -rf docs/* && typedoc",
+        "docs": "rm -rf ./../gh-pages/token-lending/* && typedoc --out ../../gh-pages/token-lending",
```
Comment (jordansexton) [inline token-lending/js/package.json:26]: Is this path supposed to be `../../gh-pages` like the others?

File: token/js/package.json [hunk 2]
```diff
@@ -35,6 +35,8 @@
   "scripts": {
     "build": "rollup -c",
     "build:browser-test": "rollup -c test/rollup.config.js",
+    "deploy": "yarn docs && gh-pages -d docs",
+    "docs": "rm -rf docs/* && documentation build client/token.js -f html --github -o ../../gh-pages/token --sort-order alpha",
```
Comment (jordansexton) [inline token/js/package.json:39]: FWIW, these have changed (along with the rest of this file) in master.

**Review by jordansexton — COMMENTED** [review]
Looks pretty good. One concern is that only the JS docs are being used to deploy the gh-pages here, which will miss the docs from the READMEs for other things.

For the latest `token/js` docs, we `yarn deploy` to the `token/js` subdirectory which retains everything else.

**Comment by KartikSoneji** [comment]
Hi, any updates on this?
The typedocs are lagging quite a bit behind the package.

**Pull request closed without merging**